lib/Analysis/SecretnessAnalysis/SecretnessAnalysis.h

```diff
@@ -0,0 +1,71 @@
+#ifndef LIB_ANALYSIS_SECRETNESSANALYSIS_SECRETNESSANALYSIS_H_
+#define LIB_ANALYSIS_SECRETNESSANALYSIS_SECRETNESSANALYSIS_H_
+
+#include <optional>
+
+#include "mlir/include/mlir/Analysis/DataFlow/SparseAnalysis.h"  // from @llvm-project
+#include "mlir/include/mlir/IR/Operation.h"  // from @llvm-project
+#include "mlir/include/mlir/IR/Value.h"      // from @llvm-project
+
+namespace mlir {
+namespace heir {
+
+class Secretness {
```

Comment:
This (and `class SecretnessAnalysis` further down) would really benefit from some documentation/comments (see `RotationAnalysis.h` or `TargetSlotAnalysis.h` for examples).